integration-tests/tests/di_diun.rs

```diff
@@ -183,3 +184,60 @@ async fn get_valid_iface_and_mac() -> Result<(String, String)> {
     }
     bail!("No valid iface found")
 }
+
+#[tokio::test]
+async fn test_device_credentials_with_tpm() -> Result<()> {
+    let ci = env::var("FDO_PRIVILEGED").is_ok();
+    if !ci {
+        L.l("Skipped test_device_credentials with TPM\nTo run this tet set env variable FDO_PRIVILEGED and run test as superuser");
+        return Ok(());
+    }
+    if std::fs::File::open("/dev/tpm0").is_ok() {
+        env::set_var("TEST_TCTI", "device:/dev/tpm0");
+    } else {
+        L.l("Skipped test_device_credentials with TPM\nTo run this test we need /dev/tpm0");
```

Comment:
Should we be testing with tpmrmX or is this test explicitly for the abrmd and we expand/add another test for the kernel RM?

Reply:
I haven't added the tpmrmX test since it doesn't work when we are using containers (as we do in Github's CI), it changes the permissions to the following:
```
crw-rw----.  1 nobody nobody  10,   224 Jul 21 08:57 tpm0
crw-rw----.  1 nobody nobody 253, 65536 Jul 21 08:57 tpmrm
```
which gives errors. 

It does work if you are using a VM and add a TPM device, but once you are using containers it breaks.
Hence, just doing a test with the software TPM to at least have something.